src/gpu/nvidia/cudnn_conv_inner_product_impl.hpp

```diff
@@ -280,12 +280,38 @@ struct cudnn_conv_inner_product_fwd_impl_t
         // cudnnGetConvolutionForwardAlgorithm for int8 type
         if (pd->src_md()->data_type != data_type::s8
                 && pd->weights_md(0)->data_type != data_type::s8) {
-            cudnnConvolutionFwdPreference_t algo_pref
-                    = CUDNN_CONVOLUTION_FWD_PREFER_FASTEST;
+            int num_algos = 0, returned_algo_count = 0;
+            CHECK(CUDNN_EXECUTE_FUNC_S(
+                    cudnnGetConvolutionForwardAlgorithmMaxCount, handle,
+                    &num_algos));
+            std::vector<cudnnConvolutionFwdAlgoPerf_t> perf_results(num_algos);
 
-            CHECK(CUDNN_EXECUTE_FUNC_S(cudnnGetConvolutionForwardAlgorithm,
+            CHECK(CUDNN_EXECUTE_FUNC_S(cudnnGetConvolutionForwardAlgorithm_v7,
                     handle, tensor_descs_[io::src], filter_desc_, conv_desc_,
-                    tensor_descs_[io::dst], algo_pref, 0, &algo_));
+                    tensor_descs_[io::dst], num_algos, &returned_algo_count,
+                    &perf_results[0]));
+
+            size_t free_memory, total_memory;
+            CHECK(CUDA_EXECUTE_FUNC_S(
+                    cuMemGetInfo, &free_memory, &total_memory));
+
+            // perf_results are sorted ascending by compute time, so the first suitable
+            // algorithm found is the one with best performance
+            bool found_algo = false;
+            for (int i = 0; !found_algo && i < returned_algo_count; i++) {
```

Comment:
```suggestion
            for (int i = 0; i < returned_algo_count; i++) {
```
And when the alg found, simply do `break;`. This applies here and below.